localstack-core/localstack/utils/docker_utils.py

```diff
@@ -147,23 +146,22 @@ def container_ports_can_be_bound(
         port_mappings.add(port.port, port.port, protocol=port.protocol)
     try:
         result = DOCKER_CLIENT.run_container(
-            _get_ports_check_docker_image(),
-            entrypoint="sh",
-            command=["-c", "echo test123"],
+            "bash",
+            command=["echo", "test123"],
             ports=port_mappings,
             remove=True,
         )
     except Exception as e:
-        if "port is already allocated" not in str(e) and "address already in use" not in str(e):
+        if "address already in use" not in str(e):
             LOG.warning(
-                "Unexpected error when attempting to determine container port status: %s", e
+                "Unexpected error when attempting to determine container port status", exc_info=e
             )
         return False
     # TODO(srw): sometimes the command output from the docker container is "None", particularly when this function is
     #  invoked multiple times consecutively. Work out why.
     if to_str(result[0] or "").strip() != "test123":
         LOG.warning(
-            "Unexpected output when attempting to determine container port status: %s", result[0]
```

Comment:
`result` is a tuple of `(stdout, stderr)` - it makes sense to log both if anything goes wrong